lib/horde/registry_impl.ex

```diff
@@ -121,26 +142,35 @@ defmodule Horde.RegistryImpl do
   def handle_info({:members_updated, reply_to}, state) do
     members = DeltaCrdt.CausalCrdt.read(state.members_pid, 30_000)
 
+    dirty_partition =
+      state.dirty_partition ||
+        Enum.any?(members, fn
+          {_node_id, %{dirty_partition: true}} -> true
+          _ -> false
+        end)
+
     member_pids =
-      MapSet.new(members, fn {_key, {members_pid, _registry_pid, _keys_pid}} ->
+      MapSet.new(members, fn {_key, %{members_pid: members_pid}} ->
         members_pid
       end)
       |> MapSet.delete(nil)
 
+    monitor_members(members, state)
```

Comment:
because of the name I thought this would start monitoring all members every time the members updated occured. perhaps `monitor_new_members` is a more clear name?

Reply:
It is clearer. Will change.